src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -788,30 +797,108 @@ private String getRouterId() {
+        manifest.graphObjUrl = getS3GraphURI();
+        manifest.jarFile = String.format("/opt/%s/%s", getTripPlannerString(), jarName);
+        manifest.jarUrl = s3JarUrl;
+        // This must be added here because logging starts immediately before defaults are set while validating the
+        // manifest
+        manifest.otpRunnerLogFile = "/var/log/otp-runner.log";
+        manifest.prefixLogUploadsWithInstanceId = true;
+        manifest.serverStartupTimeoutSeconds = 3300;
+        manifest.s3UploadBucket = getS3FolderURI().toString();
+        manifest.statusFileLocation = String.format("%s/%s", webDir, OTP_RUNNER_STATUS_FILE);
+        manifest.uploadOtpRunnerLogs = true;
+        if (!graphAlreadyBuilt) {
+            // settings when graph building needs to happen
+            manifest.buildConfigJSON = deployment.generateBuildConfigAsString();
+            manifest.buildGraph = true;
+            try {
+                manifest.gtfsAndOsmUrls = deployment.generateGtfsAndOsmUrls();
+            } catch (MalformedURLException e) {
+                status.fail("Failed to create OSM download url!", e);
                 return null;
             }
-        } catch (IOException e) {
-            status.fail(String.format("Error checking for trip planner jar: %s", s3JarUrl), e);
+            manifest.uploadGraph = true;
+            manifest.uploadGraphBuildLogs = true;
+            manifest.uploadGraphBuildReport = true;
+            if (deployment.buildGraphOnly || otpServer.ec2Info.hasSeparateGraphBuildConfig()) {
+                // This instance should be ran to only build the graph
+                manifest.runServer = false;
+            } else {
+                // This instance will both run a graph and start the OTP server
+                manifest.routerConfigJSON = deployment.generateRouterConfigAsString();
+                manifest.runServer = true;
+                manifest.uploadServerStartupLogs = true;
+            }
+        } else {
+            // This instance will only start the OTP server with a prebuilt graph
+            manifest.buildGraph = false;
+            manifest.routerConfigJSON = deployment.generateRouterConfigAsString();
+            manifest.runServer = true;
+            manifest.uploadServerStartupLogs = true;
+        }
+        String otpRunnerManifestFile = String.format("/var/%s/otp-runner-manifest.json", getTripPlannerString());
+
+        //////////////// BEGIN USER DATA
+        List<String> lines = new ArrayList<>();
+        lines.add("#!/bin/bash");
+        // NOTE: user data output is logged to `/var/log/cloud-init-output.log` automatically with ec2 instances
+        // Add some items to the $PATH as the $PATH with user-data scripts differs from the ssh $PATH.
+        lines.add("export PATH=\"$PATH:/home/ubuntu/.yarn/bin\"");
+        lines.add("export PATH=\"$PATH:/home/ubuntu/.nvm/versions/node/v12.16.3/bin\"");
```

Comment:
Should the node version be hard-coded like this?

Reply:
When adding it to the PATH variable, yes.

Reply:
@evansiroky, I'm more so asking should we bump the node version out into its own variable (and then string formatting the export PATH line) just for visibility and easy access.